include/swift/SIL/SILBuilder.h

```diff
@@ -2527,6 +2527,21 @@ class SILBuilderWithScope : public SILBuilder {
   /// predecessor block: the parent of \p inst.
   static void insertAfter(SILInstruction *inst,
                           function_ref<void(SILBuilder &)> func);
+
+  /// If \p is an inst, then this is equivalent to insertAfter(inst). If a
+  /// SILArgument is passed in, we use the first instruction in its parent
+  /// block. We assert on undef.
+  static void insertAfter(SILValue value,
+                          function_ref<void(SILBuilder &)> func) {
+    if (auto *i = dyn_cast<SingleValueInstruction>(value))
+      return insertAfter(i, func);
+    if (auto *mvir = dyn_cast<MultipleValueInstructionResult>(value))
+      return insertAfter(mvir->getParent(), func);
+    if (auto *arg = dyn_cast<SILArgument>(value))
+      return insertAfter(&*arg->getParent()->begin(), func);
+    assert(!isa<SILUndef>(value) && "This API can not use undef");
+    llvm_unreachable("Unhandled case?!");
+  }
```

Comment:
I added a similar api recently: https://github.com/apple/swift/blob/244442939dc6ed5aa8ce57203fce8147c203a30b/lib/SILOptimizer/Utils/InstOptUtils.cpp#L72 

We may want to keep just one.

Reply:
I think they are both useful in their own way. If you want I can in a follow on commit fixup the usages of getInsertAfterPoint to use this instead. Hows that sound?

Reply:
Sounds good